### PR TITLE
Adding the logger package. This creates a wrapper around winston.js

### PR DESCRIPTION
**Pull request by ramitha**

### CODE REVIEW

File: packages/@icon-magic/logger/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  error: Function;
+  debug: Function;
+  info: Function;
+}
+
+/**
+ * Wrapper around winston's logger that takes in a fileName
+ * @param fileName name of the file that's added as a label to the logging data
+ */
+export const logger = (fileName: string): Logger => {
```
Comment (zimmi88) [inline packages/@icon-magic/logger/src/index.ts]: `export function`?

File: packages/@icon-magic/logger/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    new transports.File({ filename: 'icon-magic-error.log', level: 'error' })
+  ],
+  format: combine(colorize(), timestamp(), myFormat)
+});
+
+/**
+ * Wrapper around winston's logger
+ */
+export interface Logger {
+  error: Function;
```
Comment (zimmi88) [inline packages/@icon-magic/logger/src/index.ts]: These logger methods could be more specifically defined...
`error: (msg: string) => void`

File: packages/@icon-magic/logger/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+---
+
+This package is responsible for exposing all the logging functionality for
+icon-magic. It contains a wrapper around winston.js for including the filename
+from where the logs are generated as a label in the logged message.
+
+Logs are written to the following simulataneously:
+
+1. console - logs of all levels
```
Comment (thousand) [inline packages/@icon-magic/logger/README.md]: Maybe a little out of scope for right now — any thoughts on making this configurable?

Comment (ramitha) [reply packages/@icon-magic/logger/README.md]: Yes, I've created an issue to track this: https://github.com/linkedinlabs/icon-magic/issues/14

**Pull request merged**